src/core/reactor.cc

```diff
@@ -2218,6 +2219,35 @@ void reactor::kill(pid_t pid, int sig) {
     ret.throw_if_error();
 }
 
+future<std::optional<struct group>> reactor::getgrnam(std::string_view name, char *buf, size_t buflen) {
+    syscall_result_extra<std::optional<struct group>> sr = co_await _thread_pool->submit<syscall_result_extra<std::optional<struct group>>>(
```

Comment:
for using `co_await` operator, please `#include <coroutine>`. please put it right before `#include <exception>`. so that they are sorted in alphabetical order.